### PR TITLE
Add `ExecStatements` function

### PR DESCRIPTION
**Pull request by nathan-artie**

### CODE REVIEW

File: lib/destination/dwh.go [hunk 1]
```diff
@@ -35,3 +37,37 @@ type Baseline interface {
...
+		return fmt.Errorf("statements is empty")
+	case 1:
+		slog.Debug("Executing...", slog.String("query", statements[0]))
+		if _, err := dwh.Exec(statements[0]); err != nil {
+			return fmt.Errorf("failed to execute statement: %w", err)
+		}
+
+		return nil
+	default:
+		tx, err := dwh.Begin()
```
Comment (Tang8330) [inline lib/destination/dwh.go:55]: What about rolling back tx if it failed?

Comment (nathan-artie) [reply lib/destination/dwh.go:55]: I can add that, I was just duplicating what we were already doing for Redshift merges https://github.com/artie-labs/transfer/blob/e275e192b8a886f5d8c4498da964ce37b85dd062/clients/shared/merge.go#L147-L168

Comment (Tang8330) [reply lib/destination/dwh.go:55]: Yeah, we should add it. We're also inconsistent about rolling it back.

File: lib/destination/dwh.go [hunk 2]
```diff
@@ -35,3 +37,46 @@ type Baseline interface {
...
+		var committed bool
```
Comment (Tang8330) [inline lib/destination/dwh.go:56]: nit: bring committed right before the defer

File: lib/destination/dwh.go [hunk 3]
```diff
@@ -35,3 +37,46 @@ type Baseline interface {
...
+		if err != nil {
+			return fmt.Errorf("failed to start tx: %w", err)
+		}
+		defer func() {
+			if !committed {
+				if err = tx.Rollback(); err != nil {
```
Comment (Tang8330) [inline lib/destination/dwh.go:62]: Use a different variable? This might shadow the actual `err`

**Review by Tang8330 — APPROVED** [review]
lgtm % c

**Pull request merged**